models/issue.go

```diff
@@ -1498,3 +1507,50 @@ func updateIssue(e Engine, issue *Issue) error {
 func UpdateIssue(issue *Issue) error {
 	return updateIssue(x, issue)
 }
+
+// UpdateIssueDeadline updates an issue deadline and adds comments. Setting a deadline to 0 means deleting it.
+func UpdateIssueDeadline(issue *Issue, doer *User) (err error) {
+
+	// Check if the new date was added or modified
+	var actualIssue Issue
+	_, err = x.Table("issue").
+		Select("deadline_unix").
+		ID(issue.ID).
+		Get(&actualIssue)
+	if err != nil {
+		return fmt.Errorf("getActualIssue: %v", err)
+	}
+
+	sess := x.NewSession()
+	defer sess.Close()
+
+	if err = issue.loadRepo(sess); err != nil {
+		return fmt.Errorf("loadRepo: %v", err)
+	}
+
+	// Make the comment
+	if issue.DeadlineUnix == 0 {
```

Comment:
We can move this if/else to `createDeadlineComments` to be consistent (no other create*Comment function takes CommentType as argument)
`createDeadlineComment(sess, doer, repo, issue, issue.DeadlineUnix, deadlineUnix)`

Reply:
Done.